internal/catalogmetadata/cache/cache.go

```diff
@@ -64,12 +68,18 @@ func (fsc *filesystemCache) Put(catalogName, resolvedRef string, source io.Reade
 	if errToCache == nil {
 		cacheFS, errToCache = fsc.writeFS(catalogName, source)
 	}
-	fsc.cacheDataByCatalogName[catalogName] = cacheData{
+	fsc.cacheDataByCatalogName.Store(catalogName, cacheData{
 		Ref:   resolvedRef,
 		Error: errToCache,
+	})
+
+	if errToCache != nil {
+		fsc.logger.Error(errToCache, "Cache update failed", "catalog", catalogName, "ref", resolvedRef)
+		return nil, errToCache
 	}
 
-	return cacheFS, errToCache
+	fsc.logger.Info("Cache updated successfully", "catalog", catalogName, "ref", resolvedRef)
```

Comment:
Do we have any particular concerns with the need for more visibility into the inner workings of what is going on inside this cache that motivates the extra logging?

This feels like it could start to make our logs pretty noisy. I'd suggest setting this to log at level 4 to keep our standard logging a little bit leaner.